### ISSUE TITLE
LayoutBuilder sometimes gives zero maxWidth/maxHeight on first frame

### ISSUE THREAD

**Issue by rorystephenson** [flutter/flutter#125796]
### Is there an existing issue for this?

- [X] I have searched the [existing issues](https://github.com/flutter/flutter/issues)
- [X] I have read the [guide to filing a bug](https://flutter.dev/docs/resources/bug-reports)

### Steps to reproduce

Simply run provided code in profile mode. Sometimes multiple runs are required.

There appears to be a race condition because in a larger app the issue does not *always* occur. Similarly when running in debug mode it is less common, presumably because the code runs slower. Note in the logs below that the stacktrace indicates that when the issue occurs it is part of a warm up frame.

### Expected results

Always non-zero maximum width/height in constraints from LayoutBuilder unless the parent is actually passing down zero maxWidth/maxHeight.

### Actual results

The maximum constraints are initially zero, before being updated to a correct non-zero value.

### Code sample

<details><summary>Code sample</summary>

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  @override
  Widget build(BuildContext context) {
    return const MaterialApp(home: MyHomePage());
  }
}

class MyHomePage extends StatelessWidget {
  const MyHomePage({super.key});

  @override
  Widget build(BuildContext context) {
    return LayoutBuilder(
      builder: (context, constraints) {
        if (constraints.maxHeight == 0 || constraints.maxWidth == 0) {
          throw 'Zero maximum constraints';
        }

        return Text('${constraints.maxWidth}x${constraints.maxHeight}');
      },
    );
  }
}

```

</details>

### Logs

<details><summary>Logs</summary>

```console
Launching lib/main.dart on ONEPLUS A6000 in profile mode...
Running Gradle task 'assembleProfile'...                           50.3s
✓  Built build/app/outputs/flutter-apk/app-profile.apk (11.6MB).
Installing build/app/outputs/flutter-apk/app-profile.apk...         4.2s
I/flutter (11519): BoxConstraints(w=0.0, h=0.0)
I/flutter (11519): #0      MyHomePage.build.<anonymous closure> (package:layout_builder_test/main.dart:31)
I/flutter (11519): #1      _LayoutBuilderElement._layout.layoutCallback (package:flutter/src/widgets/layout_builder.dart:119)
I/flutter (11519): #2      BuildOwner.buildScope (package:flutter/src/widgets/framework.dart:2682)
I/flutter (11519): #3      _LayoutBuilderElement._layout (package:flutter/src/widgets/layout_builder.dart:153)
I/flutter (11519): #4      RenderObject.invokeLayoutCallback.<anonymous closure> (package:flutter/src/rendering/object.dart:2300)
I/flutter (11519): #5      PipelineOwner._enableMutationsToDirtySubtrees (package:flutter/src/rendering/object.dart:1056)
I/flutter (11519): #6      RenderObject.invokeLayoutCallback (package:flutter/src/rendering/object.dart:2300)
I/flutter (11519): #7      RenderConstrainedLayoutBuilder.rebuildIfNecessary (package:flutter/src/widgets/layout_builder.dart:228)
I/flutter (11519): #8      _RenderLayoutBuilder.performLayout (package:flutter/src/widgets/layout_builder.dart:313)
I/flutter (11519): #9      RenderObject.layout (package:flutter/src/rendering/object.dart:2189)
I/flutter (11519): #10     RenderBox.layout (package:flutter/src/rendering/box.dart:2430)
I/flutter (11519): #11     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:120)
I/flutter (11519): #12     RenderObject.layout (package:flutter/src/rendering/object.dart:2189)
I/flutter (11519): #13     RenderBox.layout (package:flutter/src/rendering/box.dart:2430)
I/flutter (11519): #14     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:120)
I/flutter (11519): #15     RenderObject.layout (package:flutter/src/rendering/object.dart:2189)
I/flutter (11519): #16     RenderBox.layout (package:flutter/src/rendering/box.dart:2430)
I/flutter (11519): #17     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:120)
I/flutter (11519): #18     RenderObject.layout (package:flutter/src/rendering/object.dart:2189)
I/flutter (11519): #19     RenderBox.layout (package:flutter/src/rendering/box.dart:2430)
I/flutter (11519): #20     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:120)
I/flutter (11519): #21     RenderObject.layout (package:flutter/src/rendering/object.dart:2189)
I/flutter (11519): #22     RenderBox.layout (package:flutter/src/rendering/box.dart:2430)
I/flutter (11519): #23     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:120)
I/flutter (11519): #24     RenderObject.layout (package:flutter/src/rendering/object.dart:2189)
I/flutter (11519): #25     RenderBox.layout (package:flutter/src/rendering/box.dart:2430)
I/flutter (11519): #26     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:120)
I/flutter (11519): #27     RenderObject.layout (package:flutter/src/rendering/object.dart:2189)
I/flutter (11519): #28     RenderBox.layout (package:flutter/src/rendering/box.dart:2430)
I/flutter (11519): #29     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:120)
I/flutter (11519): #30     RenderObject.layout (package:flutter/src/rendering/object.dart:2189)
I/flutter (11519): #31     RenderBox.layout (package:flutter/src/rendering/box.dart:2430)
I/flutter (11519): #32     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:120)
I/flutter (11519): #33     RenderObject.layout (package:flutter/src/rendering/object.dart:2189)
I/flutter (11519): #34     RenderBox.layout (package:flutter/src/rendering/box.dart:2430)
I/flutter (11519): #35     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:120)
I/flutter (11519): #36     RenderObject.layout (package:flutter/src/rendering/object.dart:2189)
I/flutter (11519): #37     RenderBox.layout (package:flutter/src/rendering/box.dart:2430)
I/flutter (11519): #38     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:120)
I/flutter (11519): #39     RenderOffstage.performLayout (package:flutter/src/rendering/proxy_box.dart:3815)
I/flutter (11519): #40     RenderObject.layout (package:flutter/src/rendering/object.dart:2189)
I/flutter (11519): #41     RenderBox.layout (package:flutter/src/rendering/box.dart:2430)
I/flutter (11519): #42     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:120)
I/flutter (11519): #43     RenderObject.layout (package:flutter/src/rendering/object.dart:2189)
I/flutter (11519): #44     RenderBox.layout (package:flutter/src/rendering/box.dart:2430)
I/flutter (11519): #45     _RenderTheatre.performLayout (package:flutter/src/widgets/overlay.dart:856)
I/flutter (11519): #46     RenderObject.layout (package:flutter/src/rendering/object.dart:2189)
I/flutter (11519): #47     RenderBox.layout (package:flutter/src/rendering/box.dart:2430)
I/flutter (11519): #48     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:120)
I/flutter (11519): #49     RenderObject.layout (package:flutter/src/rendering/object.dart:2189)
I/flutter (11519): #50     RenderBox.layout (package:flutter/src/rendering/box.dart:2430)
I/flutter (11519): #51     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:120)
I/flutter (11519): #52     RenderObject.layout (package:flutter/src/rendering/object.dart:2189)
I/flutter (11519): #53     RenderBox.layout (package:flutter/src/rendering/box.dart:2430)
I/flutter (11519): #54     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:120)
I/flutter (11519): #55     RenderObject.layout (package:flutter/src/rendering/object.dart:2189)
I/flutter (11519): #56     RenderBox.layout (package:flutter/src/rendering/box.dart:2430)
I/flutter (11519): #57     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:120)
I/flutter (11519): #58     RenderObject.layout (package:flutter/src/rendering/object.dart:2189)
I/flutter (11519): #59     RenderBox.layout (package:flutter/src/rendering/box.dart:2430)
I/flutter (11519): #60     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:120)
I/flutter (11519): #61     RenderObject.layout (package:flutter/src/rendering/object.dart:2189)
I/flutter (11519): #62     RenderBox.layout (package:flutter/src/rendering/box.dart:2430)
I/flutter (11519): #63     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:120)
I/flutter (11519): #64     RenderObject.layout (package:flutter/src/rendering/object.dart:2189)
I/flutter (11519): #65     RenderBox.layout (package:flutter/src/rendering/box.dart:2430)
I/flutter (11519): #66     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:120)
I/flutter (11519): #67     RenderObject.layout (package:flutter/src/rendering/object.dart:2189)
I/flutter (11519): #68     RenderBox.layout (package:flutter/src/rendering/box.dart:2430)
I/flutter (11519): #69     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:120)
I/flutter (11519): #70     RenderObject.layout (package:flutter/src/rendering/object.dart:2189)
I/flutter (11519): #71     RenderBox.layout (package:flutter/src/rendering/box.dart:2430)
I/flutter (11519): #72     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:120)
I/flutter (11519): #73     RenderObject.layout (package:flutter/src/rendering/object.dart:2189)
I/flutter (11519): #74     RenderBox.layout (package:flutter/src/rendering/box.dart:2430)
I/flutter (11519): #75     RenderView.performLayout (package:flutter/src/rendering/view.dart:175)
I/flutter (11519): #76     RenderObject._layoutWithoutResize (package:flutter/src/rendering/object.dart:2027)
I/flutter (11519): #77     PipelineOwner.flushLayout (package:flutter/src/rendering/object.dart:1020)
I/flutter (11519): #78     RendererBinding.drawFrame (package:flutter/src/rendering/binding.dart:516)
I/flutter (11519): #79     WidgetsBinding.drawFrame (package:flutter/src/widgets/binding.dart:865)
I/flutter (11519): #80     RendererBinding._handlePersistentFrameCallback (package:flutter/src/rendering/binding.dart:381)
I/flutter (11519): #81     SchedulerBinding._invokeFrameCallback (package:flutter/src/scheduler/binding.dart:1289)
I/flutter (11519): #82     SchedulerBinding.handleDrawFrame (package:flutter/src/scheduler/binding.dart:1218)
I/flutter (11519): #83     SchedulerBinding.scheduleWarmUpFrame.<anonymous closure> (package:flutter/src/scheduler/binding.dart:942)
I/flutter (11519): #87     _RawReceivePort._handleMessage (dart:isolate-patch/isolate_patch.dart:192)
I/flutter (11519): (elided 3 frames from class _Timer and dart:async-patch)

```

</details>

### Flutter Doctor output

<details><summary>Doctor output</summary>

```console
[✓] Flutter (Channel stable, 3.7.12, on macOS 12.5.1 21G83 darwin-arm64, locale en-NZ)
    • Flutter version 3.7.12 on channel stable at /Users/rorystephenson/sdk/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 4d9e56e694 (2 weeks ago), 2023-04-17 21:47:46 -0400
    • Engine revision 1a65d409c7
    • Dart version 2.19.6
    • DevTools version 2.20.1

[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.1)
    • Android SDK at /Users/rorystephenson/sdk/android
    • Platform android-33, build-tools 33.0.1
    • ANDROID_HOME = /Users/rorystephenson/sdk/android
    • Java binary at: /Applications/Android Studio.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.15+0-b2043.56-8887301)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.2)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14C18
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2022.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.15+0-b2043.56-8887301)

[✓] Connected device (3 available)
    • ONEPLUS A6000 (mobile) • 192.168.59.56:5555 • android-arm64  • Android 11 (API 30)
    • macOS (desktop)        • macos              • darwin-arm64   • macOS 12.5.1 21G83 darwin-arm64
    • Chrome (web)           • chrome             • web-javascript • Google Chrome 111.0.5563.110

[✓] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!
```

</details>

**Comment by JaffaKetchup** [comment]
IMO, the recursive/delay based solutions found in #25827 are not acceptable. The Flutter engine should not start running the app until the available size is known from the OS/platform.

**Comment by JaffaKetchup** [comment]
@thachnb85 That appears to be a slightly different issue, I'd recommend opening a new issue, as recommended by the Flutter Contributing guidelines.

**Comment by darshankawar** [comment]
Thanks for the report. I was able to replicate the error by running the code sample in profile mode and in second attempt, it triggered the log. While doing so, I observed that it takes a bit more time in profile mode to start the app as compared to in debug or in release mode wherein, this error doesn't occur.

<details>
<summary> console error log </summary>

```

Launching lib/main.dart on Darshan's iphone in profile mode...
Automatically signing iOS for device deployment using specified development team in Xcode project: X8NNQ9CYL2
Running Xcode build...                                                  
 └─Compiling, linking and signing...                         4.3s
Xcode build done.                                           10.3s
10.7.0 - [FirebaseCore][I-COR000005] No app has been configured yet.
flutter: Zero maximum constraints
flutter: #0      MyHomePage.build.<anonymous closure> (package:flutter_app/main.dart:24)
flutter: #1      _LayoutBuilderElement._layout.layoutCallback (package:flutter/src/widgets/layout_builder.dart:119)
flutter: #2      BuildOwner.buildScope (package:flutter/src/widgets/framework.dart:2682)
flutter: #3      _LayoutBuilderElement._layout (package:flutter/src/widgets/layout_builder.dart:153)
flutter: #4      RenderObject.invokeLayoutCallback.<anonymous closure> (package:flutter/src/rendering/object.dart:2300)
flutter: #5      PipelineOwner._enableMutationsToDirtySubtrees (package:flutter/src/rendering/object.dart:1056)

flutter: #6      RenderObject.invokeLayoutCallback (package:flutter/src/rendering/object.dart:2300)
flutter: #7      RenderConstrainedLayoutBuilder.rebuildIfNecessary (package:flutter/src/widgets/layout_builder.dart:228)
flutter: #8      _RenderLayoutBuilder.performLayout (package:flutter/src/widgets/layout_builder.dart:313)
flutter: #9      RenderObject.layout (package:flutter/src/rendering/object.dart:2189)
flutter: #10     RenderBox.layout (package:flutter/src/rendering/box.dart:2430)
flutter: #11     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:120)
flutter: #12     RenderObject.layout (package:flutter/src/rendering/object.dart:2189)
flutter: #13     RenderBox.layout (package:flutter/src/rendering/box.dart:2430)
flutter: #14     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:120)
flutter: #15     RenderObject.layout (package:flutter/src/rendering/object.dart:2189)
flutter: #16     RenderBox.layout (package:flutter/src/rendering/box.dart:2430)
flutter: #17     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:120)
flutter: #18     RenderObject.layout (package:flutter/src/rendering/object.dart:2189)
flutter: #19     RenderBox.layout (package:flutter/src/rendering/box.dart:2430)
flutter: #20     ChildLayoutHelper.layoutChild (package:flutter/src/rendering/layout_helper.dart:56)
flutter: #21     RenderStack._computeSize (package:flutter/src/rendering/stack.dart:595)
flutter: #22     RenderStack.performLayout (package:flutter/src/rendering/stack.dart:622)
flutter: #23     RenderObject.layout (package:flutter/src/rendering/object.dart:2189)
flutter: #24     RenderBox.layout (package:flutter/src/rendering/box.dart:2430)
flutter: #25     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:120)
flutter: #26     RenderObject.layout (package:flutter/src/rendering/object.dart:2189)
flutter: #27     RenderBox.layout (package:flutter/src/rendering/box.dart:2430)
flutter: #28     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:120)
flutter: #29     RenderObject.layout (package:flutter/src/rendering/object.dart:2189)
flutter: #30     RenderBox.layout (package:flutter/src/rendering/box.dart:2430)
flutter: #31     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:120)
flutter: #32     RenderObject.layout (package:flutter/src/rendering/object.dart:2189)
flutter: #33     RenderBox.layout (package:flutter/src/rendering/box.dart:2430)
flutter: #34     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:120)
flutter: #35     RenderObject.layout (package:flutter/src/rendering/object.dart:2189)
flutter: #36     RenderBox.layout (package:flutter/src/rendering/box.dart:2430)
flutter: #37     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:120)
flutter: #38     RenderObject.layout (package:flutter/src/rendering/object.dart:2189)
flutter: #39     RenderBox.layout (package:flutter/src/rendering/box.dart:2430)
flutter: #40     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:120)
flutter: #41     RenderOffstage.performLayout (package:flutter/src/rendering/proxy_box.dart:3815)
flutter: #42     RenderObject.layout (package:flutter/src/rendering/object.dart:2189)
flutter: #43     RenderBox.layout (package:flutter/src/rendering/box.dart:2430)
flutter: #44     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:120)
flutter: #45     RenderObject.layout (package:flutter/src/rendering/object.dart:2189)
flutter: #46     RenderBox.layout (package:flutter/src/rendering/box.dart:2430)
flutter: #47     _RenderTheatre.performLayout (package:flutter/src/widgets/overlay.dart:856)
flutter: #48     RenderObject.layout (package:flutter/src/rendering/object.dart:2189)
flutter: #49     RenderBox.layout (package:flutter/src/rendering/box.dart:2430)
flutter: #50     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:120)
flutter: #51     RenderObject.layout (package:flutter/src/rendering/object.dart:2189)
flutter: #52     RenderBox.layout (package:flutter/src/rendering/box.dart:2430)
flutter: #53     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:120)
flutter: #54     RenderObject.layout (package:flutter/src/rendering/object.dart:2189)
flutter: #55     RenderBox.layout (package:flutter/src/rendering/box.dart:2430)
flutter: #56     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:120)
flutter: #57     RenderObject.layout (package:flutter/src/rendering/object.dart:2189)
flutter: #58     RenderBox.layout (package:flutter/src/rendering/box.dart:2430)
flutter: #59     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:120)
flutter: #60     RenderObject.layout (package:flutter/src/rendering/object.dart:2189)
flutter: #61     RenderBox.layout (package:flutter/src/rendering/box.dart:2430)
flutter: #62     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:120)
flutter: #63     RenderObject.layout (package:flutter/src/rendering/object.dart:2189)
flutter: #64     RenderBox.layout (package:flutter/src/rendering/box.dart:2430)
flutter: #65     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:120)
flutter: #66     RenderObject.layout (package:flutter/src/rendering/object.dart:2189)
flutter: #67     RenderBox.layout (package:flutter/src/rendering/box.dart:2430)
flutter: #68     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:120)
flutter: #69     RenderObject.layout (package:flutter/src/rendering/object.dart:2189)
flutter: #70     RenderBox.layout (package:flutter/src/rendering/box.dart:2430)
flutter: #71     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:120)
flutter: #72     RenderObject.layout (package:flutter/src/rendering/object.dart:2189)
flutter: #73     RenderBox.layout (package:flutter/src/rendering/box.dart:2430)
flutter: #74     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:120)
flutter: #75     RenderObject.layout (package:flutter/src/rendering/object.dart:2189)
flutter: #76     RenderBox.layout (package:flutter/src/rendering/box.dart:2430)
flutter: #77     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:120)
flutter: #78     RenderObject.layout (package:flutter/src/rendering/object.dart:2189)
flutter: #79     RenderBox.layout (package:flutter/src/rendering/box.dart:2430)
flutter: #80     RenderView.performLayout (package:flutter/src/rendering/view.dart:175)
flutter: #81     RenderObject._layoutWithoutResize (package:flutter/src/rendering/object.dart:2027)
flutter: #82     PipelineOwner.flushLayout (package:flutter/src/rendering/object.dart:1020)
flutter: #83     RendererBinding.drawFrame (package:flutter/src/rendering/binding.dart:516)
flutter: #84     WidgetsBinding.drawFrame (package:flutter/src/widgets/binding.dart:865)
flutter: #85     RendererBinding._handlePersistentFrameCallback (package:flutter/src/rendering/binding.dart:381)
flutter: #86     SchedulerBinding._invokeFrameCallback (package:flutter/src/scheduler/binding.dart:1289)
flutter: #87     SchedulerBinding.handleDrawFrame (package:flutter/src/scheduler/binding.dart:1218)
flutter: #88     SchedulerBinding.scheduleWarmUpFrame.<anonymous closure> (package:flutter/src/scheduler/binding.dart:942)
flutter: #92     _RawReceivePort._handleMessage (dart:isolate-patch/isolate_patch.dart:192)
flutter: (elided 3 frames from class _Timer and dart:async-patch)


```

</details>

<details>
<summary> stable, master flutter doctor -v </summary>

```
[!] Flutter (Channel stable, 3.7.12, on macOS 12.2.1 21D62 darwin-x64, locale
    en-GB)
    • Flutter version 3.7.12 on channel stable at
      /Users/dhs/documents/fluttersdk/flutter
    ! Warning: `flutter` on your path resolves to
      /Users/dhs/Documents/Fluttersdk/flutter/bin/flutter, which is not inside
      your current Flutter SDK checkout at
      /Users/dhs/documents/fluttersdk/flutter. Consider adding
      /Users/dhs/documents/fluttersdk/flutter/bin to the front of your path.
    ! Warning: `dart` on your path resolves to
      /Users/dhs/Documents/Fluttersdk/flutter/bin/dart, which is not inside your
      current Flutter SDK checkout at /Users/dhs/documents/fluttersdk/flutter.
      Consider adding /Users/dhs/documents/fluttersdk/flutter/bin to the front
      of your path.
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 4d9e56e694 (3 days ago), 2023-04-17 21:47:46 -0400
    • Engine revision 1a65d409c7
    • Dart version 2.19.6
    • DevTools version 2.20.1
    • If those were intentional, you can disregard the above warnings; however
      it is recommended to use "git" directly to perform update checks and
      upgrades.

[!] Xcode - develop for iOS and macOS (Xcode 12.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    ! Flutter recommends a minimum Xcode version of 13.
      Download the latest version or update via the Mac App Store.
    • CocoaPods version 1.11.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] VS Code (version 1.62.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.21.0

[✓] Connected device (5 available)
    • SM G975F (mobile)       • RZ8M802WY0X • android-arm64   • Android 11 (API 30)
    • Darshan's iphone (mobile)  • 21150b119064aecc249dfcfe05e259197461ce23 •
      ios            • iOS 14.4.1 18D61
    • iPhone 12 Pro Max (mobile) • A5473606-0213-4FD8-BA16-553433949729     •
      ios            • com.apple.CoreSimulator.SimRuntime.iOS-14-3 (simulator)
    • macOS (desktop)            • macos                                    •
      darwin-x64     • Mac OS X 10.15.4 19E2269 darwin-x64
    • Chrome (web)               • chrome                                   •
      web-javascript • Google Chrome 98.0.4758.80

[✓] HTTP Host Availability
    • All required HTTP hosts are available

! Doctor found issues in 1 category.

[!] Flutter (Channel master, 3.10.0-19.0.pre.28, on macOS 12.2.1 21D62
    darwin-x64, locale en-GB)
    • Flutter version 3.10.0-19.0.pre.28 on channel master at
      /Users/dhs/documents/fluttersdk/flutter
    ! Warning: `flutter` on your path resolves to
      /Users/dhs/Documents/Fluttersdk/flutter/bin/flutter, which is not inside
      your current Flutter SDK checkout at
      /Users/dhs/documents/fluttersdk/flutter. Consider adding
      /Users/dhs/documents/fluttersdk/flutter/bin to the front of your path.
    ! Warning: `dart` on your path resolves to
      /Users/dhs/Documents/Fluttersdk/flutter/bin/dart, which is not inside your
      current Flutter SDK checkout at /Users/dhs/documents/fluttersdk/flutter.
      Consider adding /Users/dhs/documents/fluttersdk/flutter/bin to the front
      of your path.
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 707261fa05 (40 minutes ago), 2023-05-03 00:08:20 -0400
    • Engine revision 228da31894
    • Dart version 3.1.0 (build 3.1.0-66.0.dev)
    • DevTools version 2.23.1
    • If those were intentional, you can disregard the above warnings; however
      it is recommended to use "git" directly to perform update checks and
      upgrades.

[!] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /Users/dhs/Library/Android/sdk
    ✗ cmdline-tools component is missing
      Run `path/to/sdkmanager --install "cmdline-tools;latest"`
      See https://developer.android.com/studio/command-line for more details.
    ✗ Android license status unknown.
      Run `flutter doctor --android-licenses` to accept the SDK licenses.
      See https://flutter.dev/docs/get-started/install/macos#android-setup for
      more details.

[✓] Xcode - develop for iOS and macOS (Xcode 13.2.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 13C100
    • CocoaPods version 1.11.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] IntelliJ IDEA Ultimate Edition (version 2021.3.2)
    • IntelliJ at /Applications/IntelliJ IDEA.app
    • Flutter plugin version 65.1.4
    • Dart plugin version 213.7228

[✓] VS Code (version 1.62.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.29.0

[✓] Connected device (3 available)
    • Darshan's iphone (mobile) • 21150b119064aecc249dfcfe05e259197461ce23 • ios
      • iOS 15.3.1 19D52
    • macOS (desktop)           • macos                                    •
      darwin-x64     • macOS 12.2.1 21D62 darwin-x64
    • Chrome (web)              • chrome                                   •
      web-javascript • Google Chrome 109.0.5414.119

[✓] Network resources
    • All expected network resources are available.

! Doctor found issues in 1 category.
      
[!] Xcode - develop for iOS and macOS (Xcode 12.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    ! Flutter recommends a minimum Xcode version of 13.
      Download the latest version or update via the Mac App Store.
    • CocoaPods version 1.11.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] VS Code (version 1.62.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.21.0

[✓] Connected device (5 available)
    • SM G975F (mobile)       • RZ8M802WY0X • android-arm64   • Android 11 (API 30)
    • Darshan's iphone (mobile)  • 21150b119064aecc249dfcfe05e259197461ce23 •
      ios            • iOS 14.4.1 18D61
    • iPhone 12 Pro Max (mobile) • A5473606-0213-4FD8-BA16-553433949729     •
      ios            • com.apple.CoreSimulator.SimRuntime.iOS-14-3 (simulator)
    • macOS (desktop)            • macos                                    •
      darwin-x64     • Mac OS X 10.15.4 19E2269 darwin-x64
    • Chrome (web)               • chrome                                   •
      web-javascript • Google Chrome 98.0.4758.80

[✓] HTTP Host Availability
    • All required HTTP hosts are available

! Doctor found issues in 1 category.



```

</details>

**Comment by maRci002** [comment]
This value might be zero, according to the engine:

> At startup, the size of the view may not be known before Dart code runs. If this value is observed early in the application lifecycle, it may report [Size.zero].


https://github.com/flutter/engine/blob/bc8f14001d7ae522be2302c3ea6b28d460d3e2cb/lib/ui/window.dart#L116-L137

```dart
/// The dimensions of the rectangle into which the scene rendered in this view
/// will be drawn on the screen, in physical pixels.
///
/// When this changes, [PlatformDispatcher.onMetricsChanged] is called.
///
/// At startup, the size of the view may not be known before Dart code runs.
/// If this value is observed early in the application lifecycle, it may
/// report [Size.zero].
///
/// This value does not take into account any on-screen keyboards or other
/// system UI. The [padding] and [viewInsets] properties provide information
/// about how much of each side of the view may be obscured by system UI.
///
/// This value is the same as the `size` member of [physicalGeometry].
///
/// See also:
///
///  * [physicalGeometry], which reports the location of the view as well as
///    its size.
///  * [WidgetsBindingObserver], for a mechanism at the widgets layer to
///    observe when this value changes.
Size get physicalSize => _viewConfiguration.geometry.size;
```

You can monkey patch `PlatformDispatcher.onMetricsChanged`:

```dart
import 'dart:async';
import 'dart:ui';

import 'package:flutter/material.dart';

/// Make sure [WidgetsFlutterBinding.ensureInitialized] is called before this function
Future<void> _initializeView() {
  final completer = Completer<void>();

  final oldOnMetricsChanged = PlatformDispatcher.instance.onMetricsChanged!;
  PlatformDispatcher.instance.onMetricsChanged = () {
    if (_isViewInitialized && !completer.isCompleted) {
      // revert monkey patch
      PlatformDispatcher.instance.onMetricsChanged = oldOnMetricsChanged;
      completer.complete(null);
    }

    oldOnMetricsChanged();
  };

  return completer.future;
}

bool get _isViewInitialized => PlatformDispatcher.instance.views.any((v) => v.physicalSize > Size.zero);

Future<void> main() async {
  WidgetsFlutterBinding.ensureInitialized();

  if (!_isViewInitialized) {
    await _initializeView();
  }

  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  @override
  Widget build(BuildContext context) {
    return const MaterialApp(home: MyHomePage());
  }
}

class MyHomePage extends StatelessWidget {
  const MyHomePage({super.key});

  @override
  Widget build(BuildContext context) {
    return LayoutBuilder(
      builder: (context, constraints) {
        if (constraints.maxHeight == 0 || constraints.maxWidth == 0) {
          throw 'Zero maximum constraints';
        }

        return Text('${constraints.maxWidth}x${constraints.maxHeight}');
      },
    );
  }
}
```

**Comment by goderbauer** [comment]
This is working as intended. For performance reasons we run the app as quickly as possible to get everything setup. This may mean that we don't know the dimensions yet and therefore zero is reported.

**Comment by JaffaKetchup** [comment]
The differences in behaviour between release, profile, and debug mode make this hard to test and account for. Behaviour should be as consistent as possible between these modes.
